include/z64cutscene.h

```diff
@@ -3,286 +3,481 @@
 
 #include "ultra64.h"
 
-typedef struct {
-    /* 0x00 */ u16 entrance;       // entrance index upon which the cutscene should trigger
-    /* 0x02 */ u8  ageRestriction; // 0 for adult only, 1 for child only, 2 for both ages
-    /* 0x03 */ u8  flag;           // eventChkInf flag bound to the entrance cutscene
-    /* 0x04 */ void* segAddr;      // segment offset location of the cutscene
-} EntranceCutscene; // size = 0x8
+/**
+ * Special type for blocks of cutscene data, asm-processor checks
+ * arrays for CutsceneData type and converts floats within the array
+ * to their IEEE-754 representation. The array must close with };
+ * on its own line.
+ *
+ * Files that contain this type that are included in other C files
+ * must be preceded by a '#pragma asmproc recurse' qualifier to
+ * inform asm-processor that it must recursively process that include.
+ *
+ * Example:
+ * #pragma asmproc recurse
+ * #include "file.c"
+ */
+typedef union CutsceneData {
+    s32 i;
+    f32 f;
+    s16 s[2];
+    s8  b[4];
+} CutsceneData;
 
-typedef struct {
-    /* 0x00 */ s8    continueFlag;
-    /* 0x01 */ s8    cameraRoll;
-    /* 0x02 */ u16   nextPointFrame;
-    /* 0x04 */ f32   viewAngle; // in degrees
-    /* 0x08 */ Vec3s pos;
-} CutsceneCameraPoint; // size = 0x10
+typedef enum {
+    /* 0 */ CS_STATE_IDLE,
+    /* 1 */ CS_STATE_START,
+    /* 2 */ CS_STATE_RUN,
+    /* 3 */ CS_STATE_STOP,
+    /* 4 */ CS_STATE_RUN_UNSTOPPABLE
+} CutsceneState;
 
-typedef struct {
-    /* 0x00 */ Vec3f at;
-    /* 0x0C */ Vec3f eye;
-    /* 0x18 */ s16 roll;
-    /* 0x1A */ s16 fov;
-} CutsceneCameraDirection; // size = 0x1C
+typedef enum {
+    /* 0x0001 */ CS_CMD_CAM_EYE_SPLINE = 0x01,
+    /* 0x0002 */ CS_CMD_CAM_AT_SPLINE,
+    /* 0x0003 */ CS_CMD_MISC,
+    /* 0x0004 */ CS_CMD_LIGHT_SETTING,
+    /* 0x0005 */ CS_CMD_CAM_EYE_SPLINE_REL_TO_PLAYER,
+    /* 0x0006 */ CS_CMD_CAM_AT_SPLINE_REL_TO_PLAYER,
+    /* 0x0007 */ CS_CMD_CAM_EYE,
+    /* 0x0008 */ CS_CMD_CAM_AT,
+    /* 0x0009 */ CS_CMD_RUMBLE_CONTROLLER,
+    /* 0x000A */ CS_CMD_PLAYER_CUE,
+    /* 0x000B */ CS_CMD_UNIMPLEMENTED_B,
+    /* 0x000D */ CS_CMD_UNIMPLEMENTED_D = 0x0D,
+    /* 0x000E */ CS_CMD_ACTOR_CUE_1_0,
+    /* 0x000F */ CS_CMD_ACTOR_CUE_0_0,
+    /* 0x0010 */ CS_CMD_ACTOR_CUE_1_1,
+    /* 0x0011 */ CS_CMD_ACTOR_CUE_0_1,
+    /* 0x0012 */ CS_CMD_ACTOR_CUE_0_2,
+    /* 0x0013 */ CS_CMD_TEXT,
+    /* 0x0015 */ CS_CMD_UNIMPLEMENTED_15 = 0x15,
+    /* 0x0016 */ CS_CMD_UNIMPLEMENTED_16,
+    /* 0x0017 */ CS_CMD_ACTOR_CUE_0_3,
```

Comment:
What is the reasoning for skipping some cs_cmd_ values?
e.g. why isn't this this:
```suggestion
    /* 0x000A */ CS_CMD_PLAYER_CUE,
    /* 0x000E */ CS_CMD_ACTOR_CUE_1_0 = 0xE,
    /* 0x000F */ CS_CMD_ACTOR_CUE_0_0,
    /* 0x0010 */ CS_CMD_ACTOR_CUE_1_1,
    /* 0x0011 */ CS_CMD_ACTOR_CUE_0_1,
    /* 0x0012 */ CS_CMD_ACTOR_CUE_0_2,
    /* 0x0013 */ CS_CMD_TEXT,
    /* 0x0017 */ CS_CMD_ACTOR_CUE_0_3 = 0x17,
```

Reply:
currently there are a ton of unimplemented actor cues that are extracted as `CS_UNK_DATA`
every entry that has an enum created exists somewhere in a cutscene (and will use the enum value when zap is fixed to extract these as cues)